src/zenml/utils/visualization_utils.py

```diff
@@ -0,0 +1,33 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utility functions for dashboard visualizations."""
+
+
+def format_csv_visualization_as_html(csv_visualization: str) -> str:
```

Comment:
do we have / do we want to have some kind of size limit here? something something number of rows / lines etc. Perhaps a logged warning at the very least, and then above certain points maybe we remove the middle parts etc?

Reply:
Generally it would make sense but IMO it's not worth adding right now since it's only used internally anyways

Reply:
If users really make their own custom visualizations of giant CSVs we can still fix it then ;)

Reply:
What do you mean it's only used internally? Isn't this generating an HTML string that will be visualised on the dashboard at some point? If so, don't we need to take care that people don't visualise CSV files that are too large or unwieldy for the browser to handle?

Reply:
agreed with alex here

Reply:
Ok fine I added some logic to deal with too large CSVs (larger than 10x10), if there are too many rows the middle ones will be replaced by "...", if there are too any cols the last ones will be replaced by "..."